src/mscorlib/shared/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -686,6 +686,115 @@ protected EventSource(EventSourceSettings settings, params string[] traits)
             Initialize(eventSourceGuid, eventSourceName, traits);
         }
 
+#if FEATURE_PERFTRACING
+        // Generate all EventPipeEvent instances which belong to this EventProvider
+        private unsafe void GenerateEventPipeEvents()
+        {
+            Debug.Assert(m_eventData != null);
+            Debug.Assert(m_provider != null);
+            int cnt = m_eventData.Length;
+            for (int i = 0; i < cnt; i++)
+            {
+                uint eventID = (uint)m_eventData[i].Descriptor.EventId;
+                if (eventID == 0)
+                    continue;
+
+                string eventName = m_eventData[i].Name;
+                Int64 keywords = m_eventData[i].Descriptor.Keywords;
+                uint eventVersion = m_eventData[i].Descriptor.Version;
+                uint level = m_eventData[i].Descriptor.Level;
+                byte[] eventNameBytes = Encoding.ASCII.GetBytes(eventName);
```

Comment:
Actually the default format for serializing strings in ETW (and therefore EventSource), is a null terminated unicode string.  Thus you don't need to convert to ascii (you can do a 'fixed(char* x = s)).     The same will be true for parameter names below.